src/ctapipe/reco/sklearn.py

```diff
@@ -386,6 +421,53 @@ def _predict_score(self, key, table):
     def _get_positive_index(self, key):
         return np.nonzero(self._models[key].classes_ == self.positive_class)[0][0]
 
+    def write(self, path, meta={}, overwrite=False):
+        """
+        Save a dictionary using joblib-pickle, which contains all
```

Comment:
This looks like you are implementing your own way of

https://docs.python.org/3/library/pickle.html#object.__getstate__
and
https://docs.python.org/3/library/pickle.html#object.__setstate__

Reply:
I started this with the intention of completely removing pickle. I can also convert this to draft until I can finish this.